plugins/filter_ecs/ecs.c

```diff
@@ -1532,29 +1600,23 @@ static int cb_ecs_filter(const void *data, size_t bytes,
         }
 
         /* append new keys */
-        mk_list_foreach_safe(head, tmp, &ctx->metadata_keys) {
-            metadata_key = mk_list_entry(head, struct flb_ecs_metadata_key, _head);
-            val = flb_ra_translate(metadata_key->ra, NULL, 0,
-                                   metadata_buffer->obj, NULL);
-            if (!val) {
-                flb_plg_info(ctx->ins, "Translation failed for %s : %s",
-                             metadata_key->key, metadata_key->template);
-                msgpack_unpacked_destroy(&result);
-                msgpack_sbuffer_destroy(&tmp_sbuf);
-                return FLB_FILTER_NOTOUCH;
+        if (metadata_buffer->keypairs_len > 0) {
+            mk_list_foreach_safe(head, tmp, &metadata_buffer->metadata_keypairs) {
+                keypair = mk_list_entry(head, struct flb_ecs_metadata_keypair, _head);
+
+                len = flb_sds_len(keypair->key);
+                msgpack_pack_str(&tmp_pck, len);
+                msgpack_pack_str_body(&tmp_pck,
+                                    keypair->key,
+                                    len);
+                len = flb_sds_len(keypair->val);
```

Comment:
will value always exist? I noticed that in the deletion code you check if value exists before deleting it.

Reply:
see other comment, I suppose it does always exist. the code is fine, though I'm not sure why in some places   val == null is checked before use and other places it is not

Reply:
in this case, these are pre-processed metadata keypairs, so they are guaranteed to have non-null values.